src/test/java/org/mockitousage/constructor/CreatingMocksWithConstructorTest.java

```diff
@@ -52,11 +58,51 @@ public void can_spy_abstract_classes() {
     }
 
     @Test
+    public void can_spy_abstract_classes_with_constructor_args() {
+        AbstractMessage mock = mock(AbstractMessage.class, withSettings().useConstructor("hello!").defaultAnswer(CALLS_REAL_METHODS));
+        assertEquals("hello!", mock.getMessage());
+    }
+
+    @Test
+    public void can_spy_abstract_classes_with_constructor_primitive_args() {
+        AbstractMessage mock = mock(AbstractMessage.class, withSettings().useConstructor(7).defaultAnswer(CALLS_REAL_METHODS));
+        assertEquals("7", mock.getMessage());
+    }
+
+    @Test
+    public void can_spy_abstract_classes_with_constructor_array_of_nulls() {
+        AbstractMessage mock = mock(AbstractMessage.class, withSettings().useConstructor(new Object[]{null}).defaultAnswer(CALLS_REAL_METHODS));
+        assertNull(mock.getMessage());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
Can you assert on exception message? I'd like the exception message to tell the user what to do if he needs to pass null. For example:

>"If you need to pass null, please cast it to the right type, example:
useConstructor((String) null)"